server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -469,5 +477,76 @@ private static boolean canEarlyTerminate(IndexReader reader, SortAndFormats sort
         return true;
     }
 
+    /**
+    * Returns true if more than 50% of data in the index have the same value
+    * The evaluation is approximation based on finding the median value and estimating its count
+    * Returns true if the total count of median values is greater or equal to half of the total count of documents
+    */
+    static boolean indexFieldHasDuplicateData(IndexReader reader, String field) throws IOException {
+        long globalDocCount = 0;
+        long globalMedianCount = 0;
+        for (LeafReaderContext lrc : reader.leaves()) {
+            PointValues pointValues = lrc.reader().getPointValues(field);
+            if (pointValues == null) continue;
+            int docCount = pointValues.getDocCount();
```

Comment:
Can you add a comment (or an assert) that the doc count is equals to the number of points. This is important since we'll need to change the logic here if we handle multiple values per docs (https://github.com/elastic/elasticsearch/pull/43121/files#diff-ec88da77f16eaf2fff65965789ea44beR398). Or maybe you can use `PointValues#size` here and add an assert that `PointValues#size == PointValues#getDocCount` to ensure that we don't forget to revise the logic if/when we handle multiple values per doc.